src/kv/test/kv_test.cpp

```diff
@@ -131,26 +153,154 @@ TEST_CASE("Reads/writes and deletions")
     auto vc = view3->get(k);
     REQUIRE(!vc.has_value());
   }
+}
+
+TEST_CASE_TEMPLATE("foreach", MapImpl, RawMapTypes, ExperimentalMapTypes)
```

Comment:
We were missing a lot of interesting cases of `foreach`, so I've added them here.